agent/http.go

```diff
@@ -194,6 +201,12 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 			fmt.Fprint(resp, errMsg)
 		}
 
+		if s.blacklist.IsDisallowed(req.URL.Path) {
+			err := fmt.Errorf("Permission denied, endpoint is disabled")
+			handleErr(err)
```

Comment:
Yeah this is gross - I'll factor this out to make it more explicit. The other code is a little fragile but is baked in a few different places (and complicated by the fact the the errors go over the RPC interface, which launders their types into a generic error), so we can tackle that in another change.